src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -885,29 +917,37 @@ static void layoutBarLine(const BarLine* item, LayoutContext& ctx, BarLine::Layo
 
     ldata->setBbox(r);
 
+    //! NOTE The types are listed here explicitly to show what types there are (see add method)
+    //! and accordingly show what the barline layout depends on.
     for (EngravingItem* e : *item->el()) {
-        TLayout::layoutItem(e, ctx);
-        if (e->isArticulationFamily()) {
-            Articulation* a = toArticulation(e);
+        switch (e->type()) {
+        case ElementType::ARTICULATION: {
```

Comment:
Note that this is not equivalent to `isArticulationFamily()`. But I think it's fine, given `BarLine::add`.